scripts/ci-entrypoint.sh

```diff
@@ -225,7 +235,8 @@ if [[ -n "${TEST_CCM:-}" ]]; then
         --set cloudControllerManager.enableDynamicReloading="${ENABLE_DYNAMIC_RELOADING}" \
         --set cloudControllerManager.cloudConfig="${CLOUD_CONFIG}" \
         --set cloudControllerManager.cloudConfigSecretName="${CONFIG_SECRET_NAME}" \
-        --set cloudControllerManager.logVerbosity="${CCM_LOG_VERBOSITY}"
+        --set cloudControllerManager.logVerbosity="${CCM_LOG_VERBOSITY}" \
+        --set-string cloudControllerManager.clusterCIDR="${CCM_CLUSTER_CIDR}"
```

Comment:
@jackfrancis do we need to do this in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/cloud-provider-azure.go#L44 as well?

@lzhecheng what's the impact if the clusterCIDR of CCM doesn't match the Cluster's CIDR(s)?

Reply:
clusterCIDR is for route and node ipam controllers so I think these 2 components won't work properly.

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2852

Reply:
Thanks for filing the issue @CecileRobertMichon, I agree we can add this CIDR configuration to our PR E2E scenarios as a follow-up step